### DISCUSSION TITLE
RabbitMQ Stream - Consumer is not reconnecting after RabbitMQ service restarts.

### DISCUSSION THREAD

**Discussion by praveengopalan** [rabbitmq/rabbitmq-stream-dotnet-client#359] [Q&A] [answered]
### Describe the bug

I am new to rabbitmq stream, I have created two separate applications for publishing and subscribing rabbitmq stream messages. When I restart rabbitmq service the producer app reconnects successfully. However, the consumer app is not. I am getting below error message.
<img width="960" alt="image" src="https://github.com/rabbitmq/rabbitmq-server/assets/18613379/6f124257-2509-4576-a5e5-34336b232c19">

producer and consumer code attached.

[GettingStartedProducer.txt](https://github.com/rabbitmq/rabbitmq-server/files/14290262/GettingStartedProducer.txt)
[GettingStartedConsumer.txt](https://github.com/rabbitmq/rabbitmq-server/files/14290263/GettingStartedConsumer.txt)

### Reproduction steps

1. start producer and consumer app
2. stop rabbitmq service
3. start rabbitmq service
...

### Expected behavior

Consumer reconnect back

**Comment by michaelklishin** [comment]
Please put together an executable way of reproducing and share node logs. We do not guess in this community.

All that can be suggested from these screenshots is that the client did try to connect or reconnect (the output is incomplete) and that failed with

> No connection could be made … target machine actively refused it

which is TCP speak for "the connection was refused or there was no process listening on the target hostname:port".

This client [uses a back-off reconnection strategy by default](https://rabbitmq.github.io/rabbitmq-stream-dotnet-client/stable/htmlsingle/index.html#_configuring_the_stream_system), so each failed attempt will increase the interval it waits before the next attempt to reconnect. Perhaps after Nth attempt to reconnect the interval grows to tens of seconds. With an executable example anyone could quickly verify this, in particular by [taking a traffic capture](https://rabbitmq.com/amqp-wireshark.html) and observing inbound TCP connection attempts on the RabbitMQ Stream protocol port.

**Comment by Gsantomaggio** [comment, marked as answer]
> ### Describe the bug
> 
> I am new to rabbitmq stream, I have created two separate applications for publishing and subscribing rabbitmq stream messages. When I restart rabbitmq service the producer app reconnects successfully. However, the consumer app is not. I am getting below error message.
> 



We'd need also the rabbitmq version and the client version.

Yesterday we released the 1.8.0 version focus on reconnecting

**Comment by Gsantomaggio** [comment]
@praveengopalan, looking at the logs, you are using an old client version.
Please update to the last version `(1.8.0)` and try this:
https://github.com/rabbitmq/rabbitmq-stream-dotnet-client/tree/main/docs/ReliableClient

**Reply by praveengopalan** [reply]
@Gsantomaggio client version upgraded to 1.8.0 and it is working fine

**Comment by Gsantomaggio** [comment]
Ok great! 
Closed via https://github.com/rabbitmq/rabbitmq-stream-dotnet-client/discussions/359#discussioncomment-8476618
